asm/durexforth.asm

```diff
@@ -26,38 +40,38 @@ TIB = $200
 ; in separate ranges on the zeropage, so that popping and
 ; pushing gets faster (only one inx/dex operation).
 X_INIT = 0
-MSB = $73 ; high-byte stack placed in [$3b .. $72]
-LSB = $3b ; low-byte stack placed in [3 .. $3a]
-
-W = $8b ; rnd seed
-W2 = $8d ; rnd seed
-W3 = $9e ; tape error log
-
-OP_JMP = $4c
-OP_JSR = $20
-OP_RTS = $60
-OP_INX = $e8
-
-PUTCHR = $ffd2 ; put char
-
-K_RETURN = $d
-K_CLRSCR = $93
-K_SPACE = ' '
 
 ; -------- dictionary
 
-WORDLIST_BASE = $9fff
+;; Word flags
+F_IMMEDIATE = $80
+; When set, calls to the word will not be subject to tail call elimination.
+; I.e., "jsr WORD + rts" will not be replaced by "jmp WORD".
+F_NO_TAIL_CALL_ELIMINATION = $40
+STRLEN_MASK = $1f
+
 * = WORDLIST_BASE
 
 !byte 0 ; zero name length = end of dictionary.
 
+!set __LATEST = WORDLIST_BASE
+!macro BACKLINK .name , .namesize {
+    !set .xt = *
+    * = __LATEST - len(.name) - 3
+    !set __LATEST = *
+    !byte .namesize
+    !text .name
+    !word .xt
+    * = .xt
+}
```

Comment:
While moving this macro I also changed the definition. I hope it's clearer. Not that it wasn't really before.